src/Provider/TMSProvider.js

```diff
@@ -23,27 +24,33 @@ function preprocessDataLayer(layer) {
             max: 18,
         };
     }
+    layer.fx = layer.fx || 0.0;
 }
 
 function executeCommand(command) {
     const layer = command.layer;
     const tile = command.requester;
 
     const promises = [];
+
     for (const coordTMS of tile.getCoordsForLayer(layer)) {
         const coordTMSParent = (command.targetLevel < coordTMS.zoom) ?
             OGCWebServiceHelper.WMTS_WGS84Parent(coordTMS, command.targetLevel) :
             undefined;
-
         const urld = URLBuilder.xyz(coordTMSParent || coordTMS, layer);
 
-        promises.push(OGCWebServiceHelper.getColorTextureByUrl(urld, layer.networkOptions).then((texture) => {
+        const promise = layer.format === 'application/x-protobuf;type=mapbox-vector' ?
+            VectorTileHelper.getVectorTileTextureByUrl(urld, tile, layer, coordTMS) :
+            OGCWebServiceHelper.getColorTextureByUrl(urld, layer.networkOptions);
+
+        promises.push(promise.then((texture) => {
             const result = {};
-            result.texture = texture;
-            result.texture.coords = coordTMSParent || coordTMS;
-            result.pitch = coordTMSParent ?
+            const pitch = coordTMSParent ?
```

Comment:
I think all these changes can be removed, since it's equivalent to the original code.

Reply:
Yep